stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/EventsParser.kt

```diff
@@ -36,6 +39,13 @@ internal class EventsParser(
     private val logger = ChatLogger.get("Events")
     private var closedByClient = true
 
+    /**
+     * Processes typing events and "cleans" stale [TypingStartEvent]s
+     * by firing out their [TypingStopEvent] counterparts that hold the
+     * same important data, e.g. user, cid, .. etc
+     */
+    private val typingEventCache = TypingEventCache { event -> onEvent(event) }
```

Comment:
`EventsParser` should not look like a good place to keep `typingEventCache` based on its name.
Would it be possible to move `typingEventCache` into the layers above?